src/lib/list.c

```diff
@@ -335,3 +335,64 @@ struct list *list_deduplicate(struct list *list, comparison_fn_t comparison_fn,
 
 	return list;
 }
+
+struct list *list_filter_common_elements(struct list *list1, struct list *list2, comparison_fn_t comparison_fn, list_free_data_fn_t list_free_data_fn)
+{
+	/* is mandatory that list1 and list2 are sorted */
+	struct list *iter1, *iter2 = NULL;
+	struct list *preserver = NULL;
+	void *item1, *item2 = NULL;
+	int comp;
+	enum comparison_results {
```

Comment:
We can't have a comparison_fn_f function with different meanings depending on the list function that uses it. We need to be consistent. I'm still trying to understand why we need 4 different return codes and not just 3.